### PR TITLE
Correct `DynamicLibrary` lookup for iOS/macOS on DEBUG mode

### PR DESCRIPTION
**Pull request by AlexV525**
When more than one FRB plugins are introduced, their executable is not the same, which `.executable` will cause a wrong lookup with the library, and corrupt the bridge calls.

The change request corrects how libraries are located for the DEBUG mode. They will be stored at `package_name.framework/package_name`.

### CODE REVIEW

**Comment by AlexV525** [comment]
It also requires an update to the doc of FRB. FYI @fzyzcjy

**Comment by fzyzcjy** [comment]
Issue created -> https://github.com/fzyzcjy/flutter_rust_bridge/issues/1559

**Comment by Desdaemon** [comment]
Thank you for your contribution!

File: lib/ffi.dart [hunk 1]
```diff
@@ -2,21 +2,32 @@
...
-final Native api = NativeImpl(io.Platform.isIOS || io.Platform.isMacOS
-    ? DynamicLibrary.executable()
-    : DynamicLibrary.open(_dylib));
+final Native api = NativeImpl(
+  io.Platform.isIOS || io.Platform.isMacOS
+      ? _kDebugMode
+          ? DynamicLibrary.open('$_package.framework/$_package')
+          : DynamicLibrary.executable()
+      : DynamicLibrary.open(_dylib),
+);
```
Comment (fzyzcjy) [inline lib/ffi.dart:29]: I checked how I implement the v2 loader - from the comments, one reference is https://github.com/flutter/flutter/blob/8b6277e63868c2029f1e2327879b7899be44fbe2/packages/flutter_tools/templates/plugin_ffi/lib/projectName.dart.tmpl#L47-L58.

It seems that, under both debug and release mode, the symbols are from `'$_libName.framework/$_libName'`. Therefore, maybe even in release mode we should use `open`?

image

Comment (Desdaemon) [reply lib/ffi.dart:29]: Interesting. I'm not actively using Flutter on MacOS right now so I'll wait for more reports to stream in if it works or not.

Comment (fzyzcjy) [reply lib/ffi.dart:29]: I can also confirm that, using `flutter build macos` on frb_example, I generated an app and I can open and run it well on macos. It is based on v2's default loader logic, thus is using the `open` on debug and release.

Comment (AlexV525) [reply lib/ffi.dart:29]: > I checked how I implement the v2 loader - from the comments, one reference is https://github.com/flutter/flutter/blob/8b6277e63868c2029f1e2327879b7899be44fbe2/packages/flutter_tools/templates/plugin_ffi/lib/projectName.dart.tmpl#L47-L58.

I guess we should follow this then. I'll run another test for releases.

**Pull request merged**